tests/app/dao/test_provider_details_dao.py

```diff
@@ -78,3 +83,72 @@ def test_update_adds_history(restore_provider_details):
     assert not ses_history[1].active
     assert ses_history[1].version == 2
     assert ses_history[1].updated_at == datetime(2000, 1, 1, 0, 0, 0)
+
+
+def test_update_sms_provider_to_inactive_sets_inactive(restore_provider_details):
+    set_primary_sms_provider('mmg')
+    primary_provider = get_current_provider('sms')
+    primary_provider.active = False
+
+    dao_update_provider_details(primary_provider)
+
+    assert not primary_provider.active
+
+
+def test_get_current_sms_provider_returns_correct_provider(restore_provider_details):
+    set_primary_sms_provider('mmg')
+
+    provider = get_current_provider('sms')
+
+    assert provider.identifier == 'mmg'
+
+
+@pytest.mark.parametrize('provider_identifier', ['firetext', 'mmg'])
+def test_get_alternative_sms_provider_returns_expected_provider(notify_db, provider_identifier):
+    provider = get_alternative_sms_provider(provider_identifier)
+    assert provider.identifier != provider
+
+
+def test_switch_sms_provider_to_current_provider_does_not_switch(
+    notify_db,
+    notify_db_session,
```

Comment:
don't need these `notify_db*` fixtures here (or elsewhere in this file) - they're used by `restore_provider_details`